### PR TITLE
Add movielens recommendation with keras 3.0

### PR DESCRIPTION
**Pull request by madhusshivakumar**
Upgrade Keras to the Latest Version and Adapt the Code for Compatibility

In this commit, I have upgraded the Keras library to its latest version to take advantage of new features and improvements. Additionally, I have made necessary code modifications to maintain the previous functionality while utilizing the updated Keras API.

Key Changes:

Upgraded Keras to version 3.0
Updated deprecated Keras functions to their latest equivalents.
Tested the code to verify that it functions correctly with the new Keras version.
These changes aim to keep our project up to date with the latest tools and technologies while preserving the integrity of our existing functionality.

### CODE REVIEW

**Comment by madhusshivakumar** [comment]
@fchollet Please review these changes.

**Comment by divyashreepathihalli** [comment]
@madhusshivakumar thank you for the PR!
- can you post the git diff between https://github.com/keras-team/keras-io/blob/master/examples/structured_data/movielens_recommendations_transformers.py and your file?
- also can you add a colab where you verified that this example runs with no errors?
Thanks!

**Comment by madhusshivakumar** [comment]
@divyashreepathihalli thank you for your review,

Here you can find the colab notebook: 
https://colab.research.google.com/drive/1Kqm1WHruKu9QTBckp-2sF6EEJTacyFqp?usp=sharing

Here is the git diff of the changes:
*************************************************************************************************
 import numpy as np
 import pandas as pd
 import tensorflow as tf
-from tensorflow import keras
-from tensorflow.keras import layers
-from tensorflow.keras.layers import StringLookup
+import keras
+from keras import layers
+from keras.layers import StringLookup
+
 
 """
 ## Prepare the data
@@ -92,7 +93,7 @@ ratings = pd.read_csv(
 )
 
 movies = pd.read_csv(
-    "ml-1m/movies.dat", sep="::", names=["movie_id", "title", "genres"]
+    "ml-1m/movies.dat", sep="::", names=["movie_id", "title", "genres"], encoding='latin-1'
 )
 
 """
@@ -304,19 +305,19 @@ def get_dataset_from_csv(csv_file_path, shuffle=False, batch_size=128):
 
 def create_model_inputs():
     return {
-        "user_id": layers.Input(name="user_id", shape=(1,), dtype=tf.string),
+        "user_id": layers.Input(name="user_id", shape=(1,), dtype="string"),
         "sequence_movie_ids": layers.Input(
-            name="sequence_movie_ids", shape=(sequence_length - 1,), dtype=tf.string
+            name="sequence_movie_ids", shape=(sequence_length - 1,), dtype="string"
         ),
         "target_movie_id": layers.Input(
-            name="target_movie_id", shape=(1,), dtype=tf.string
+            name="target_movie_id", shape=(1,), dtype="string"
         ),
         "sequence_ratings": layers.Input(
-            name="sequence_ratings", shape=(sequence_length - 1,), dtype=tf.float32
+            name="sequence_ratings", shape=(sequence_length - 1,), dtype="float32"
         ),
-        "sex": layers.Input(name="sex", shape=(1,), dtype=tf.string),
-        "age_group": layers.Input(name="age_group", shape=(1,), dtype=tf.string),
-        "occupation": layers.Input(name="occupation", shape=(1,), dtype=tf.string),
+        "sex": layers.Input(name="sex", shape=(1,), dtype="string"),
+        "age_group": layers.Input(name="age_group", shape=(1,), dtype="string"),
+        "occupation": layers.Input(name="occupation", shape=(1,), dtype="string"),
     }
 
 
@@ -410,7 +411,7 @@ def encode_input_features(
     movie_genres_lookup = layers.Embedding(
         input_dim=genre_vectors.shape[0],
         output_dim=genre_vectors.shape[1],
-        embeddings_initializer=tf.keras.initializers.Constant(genre_vectors),
+        embeddings_initializer=keras.initializers.Constant(genre_vectors),
         trainable=False,
         name="genres_vector",
     )
@@ -447,20 +448,20 @@ def encode_input_features(
         output_dim=movie_embedding_dims,
         name="position_embedding",
     )
-    positions = tf.range(start=0, limit=sequence_length - 1, delta=1)
+    positions = keras.ops.arange(start=0, stop=sequence_length - 1, step=1)
     encodded_positions = position_embedding_encoder(positions)
     # Retrieve sequence ratings to incorporate them into the encoding of the movie.
-    sequence_ratings = tf.expand_dims(inputs["sequence_ratings"], -1)
+    sequence_ratings = keras.ops.expand_dims(inputs["sequence_ratings"], -1)
     # Add the positional encoding to the movie encodings and multiply them by rating.
     encoded_sequence_movies_with_poistion_and_rating = layers.Multiply()(
         [(encoded_sequence_movies + encodded_positions), sequence_ratings]
     )
 
     # Construct the transformer inputs.
-    for encoded_movie in tf.unstack(
+    for encoded_movie in keras.ops.unstack(
         encoded_sequence_movies_with_poistion_and_rating, axis=1
     ):
-        encoded_transformer_features.append(tf.expand_dims(encoded_movie, 1))
+        encoded_transformer_features.append(keras.ops.expand_dims(encoded_movie, 1))
     encoded_transformer_features.append(encoded_target_movie)
 
     encoded_transformer_features = layers.concatenate(
@@ -563,4 +564,4 @@ You can try training this model with different configurations, for example, by i
 the input sequence length and training the model for a larger number of epochs. In addition,
 you can try including other features like movie release year and customer
 zipcode, and including cross features like sex X genre.
-"""
+"""

*************************************************************************************************

**Review by divyashreepathihalli — APPROVED** [review]
